resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/DelayedSnapshotJob.java

```diff
@@ -28,6 +31,7 @@ public class DelayedSnapshotJob extends Job {
 	private static final String MSG_SNAPSHOT = Messages.resources_snapshot;
 	private SaveManager saveManager;
 	private Workspace workspace;
+	private boolean suspended;
```

Comment:
this must be volatile

Reply:
You might make this an integer and just increment and decrement it when `suspend` or `resume` are being called. This would relieve existing client code from checking the `isSuspended` state before resuming again. `isSuspended` might become obsolete even.

Reply:
I made it an `AtomicInteger`, thanks for the tips to you both!